### PR TITLE
Update fv3net environment to python 3.8

### PR DESCRIPTION
**Pull request by oliverwm1**
This ensures a consistent python major version between the `fv3net` and `prognostic_run` docker images. It will also let us use some nice python 3.8 features in `fv3net` (e.g. the `Literal` type is helpful for configuration!).

Requirement changes:
- Updated python dependency in `environment.yml` to 3.8
- [x] Ran `make lock_deps/lock_pip` following these [instructions](https://vulcanclimatemodeling.com/docs/fv3net/dependency_management.html#dependency-management)

### CODE REVIEW

**Comment by nbren12** [comment]
LGTM. FYI the fv3net build did make some complaints (which didn't fail). not sure if they are new or not.
- https://app.circleci.com/pipelines/github/ai2cm/fv3net/10654/workflows/20505ab9-1387-483c-bbfe-4d3993db2c9e/jobs/93203?invite=true#step-109-1045
- https://app.circleci.com/pipelines/github/ai2cm/fv3net/10654/workflows/20505ab9-1387-483c-bbfe-4d3993db2c9e/jobs/93203?invite=true#step-109-2382

File: conda-osx-64.lock [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # platform: osx-64
```
Comment (nbren12) [inline conda-osx-64.lock:1]: not sure if there are any new deps in here

**Review by nbren12 — APPROVED** [review]

File: conda-osx-64.lock (hunk 1 above)
Comment (oliverwm1) [reply conda-osx-64.lock:1]: Not as far as I see, but some of the deps got reordered which makes it a bit hard to tell.

**Comment by oliverwm1** [comment]
> LGTM. FYI the fv3net build did make some complaints (which didn't fail). not sure if they are new or not.
> 
> * https://app.circleci.com/pipelines/github/ai2cm/fv3net/10654/workflows/20505ab9-1387-483c-bbfe-4d3993db2c9e/jobs/93203?invite=true#step-109-1045
>
> * https://app.circleci.com/pipelines/github/ai2cm/fv3net/10654/workflows/20505ab9-1387-483c-bbfe-4d3993db2c9e/jobs/93203?invite=true#step-109-2382

I found both of these complaints in the fv3net build for a recent master commit 373b39524888bb784f488b6af81282de1cd3aed4: e.g. [here](https://app.circleci.com/pipelines/github/ai2cm/fv3net/10632/workflows/f4a0cc6b-c9bb-4b47-af1c-96e62dade9c4/jobs/92979?invite=true#step-109-1282) and [here](https://app.circleci.com/pipelines/github/ai2cm/fv3net/10632/workflows/f4a0cc6b-c9bb-4b47-af1c-96e62dade9c4/jobs/92979?invite=true#step-109-2648)

Probably a good idea to get on a more recent dask version though, I could address that in a separate PR.

**Pull request merged**